### PR TITLE
Closing #173 - authorization denied for builtin providers

### PR DESCRIPTION
**Pull request by autodidaddict**

### CODE REVIEW

**Comment by stevelr** [comment]
moved verify_link above the target `:unknown` check, because builtins have target unknown.
had to add authorize_link for builtins also.
These changes made both logging and numbergen calls from actors work.
Thanks for your fix this morning - it pointed me in the right direction.

**Comment by stevelr** [comment]
commit 4dc3d4d allowed the actor test cases to work, but introduced a new bug that the actor is authorized to use a builtin provider even if its wasm wasn't signed with that capability - looking into this

**Comment by stevelr** [comment]
commit f912fe5 fixes authorization for builtins that are in claims

**Review by stevelr — APPROVED** [review]
These changes partially fixed the problem and pointed me in the right direction. I added a couple more commits to this branch to address remaining issues.

**Review by brooksmtownsend — APPROVED** [review]
Clean ✅

**Comment by autodidaddict** [comment]
@stevelr changes made after the initial pull request have broken the tests. It looks like the test currently failing is invocation by call alias.

**Comment by autodidaddict** [comment]
That used to be there.

File: host_core/lib/host_core/web_assembly/imports.ex [hunk 1]
```diff
@@ -227,6 +227,11 @@ defmodule HostCore.WebAssembly.Imports do
     %{token | authorized: Enum.member?(claims.caps, namespace)}
   end
 
+  # allow actor-to-actor calls
+  defp authorize_call(token = %{target: {:actor, _, _}}) do
+    %{token | authorized: true}
+  end
```
Comment (brooksmtownsend) [inline host_core/lib/host_core/web_assembly/imports.ex:233]: So if the target is an actor, claims and namespace are irrelevant. Looks like this is what we had before 👍🏻

**Review by brooksmtownsend — APPROVED** [review]

File: host_core/lib/host_core/web_assembly/imports.ex (hunk 1 above)
Comment (autodidaddict) [reply host_core/lib/host_core/web_assembly/imports.ex:233]: Yep, at least until we support extensibility with OPA calls

**Pull request merged**